dask/array/overlap.py

```diff
@@ -686,9 +698,14 @@ def assert_int_chunksize(xs):
         assert all(type(c) is int for x in xs for cc in x.chunks for c in cc)
 
     assert_int_chunksize(args)
+    # import pdb; pdb.set_trace()
     if not trim and "chunks" not in kwargs:
         kwargs["chunks"] = args[0].chunks
-    args = [overlap(x, depth=d, boundary=b) for x, d, b in zip(args, depth, boundary)]
+    allow_rechunk = kwargs.pop("allow_rechunk", True)
```

Comment:
Per the comments, kwargs is reserved for arguments to map_blocks.  I think allow_rechunk needs to be a new argument in the map_overlap signature, because it's specific to map_overlap.

Reply:
Well, it's specific to the `overlap` function. I'm a bit reluctant to give this too much visibility, it seems like something we don't want most people to do, with a few exceptions (of which you are one).

Reply:
I agree that most won't use this.  But I don't think it good practice to hide parameters, particularly one that may change the underlying data structure.  (In addition to the point that the comments explicitly desire kwargs to be reserved for map_blocks)

As a concrete example of a problem, one cannot write the output to zarr after doing an overlap, because overlap can make the chunking uneven (which zarr does not support).  When an error crops up during the write to zarr, I think it helpful that one can look back over the methods used and easily see in a function signature that rechunking might be involved.